app/controllers/application_controller.rb

```diff
@@ -2,4 +2,50 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user, :current_order, :current_user_is_owner, :require_login
+
+  def current_user
+      @user ||= User.find_by(id: session[:user_id])
+  end
+
+  def current_user_is_owner
+    session[:user_id] == params[:user_id]
+  end
+
+  def current_order
+    # check if an order is already stored in the session
+    order ||= Order.find_by(id: session[:order_id])
+
+    # if no order exists, create a new one and store its id in sessions hash
+    if order.nil?
+      order = Order.create(order_state: "pending")
+      order.save
+      order.update(user_id: current_user.id) if current_user
+      session[:order_id] = order.id
+    end
+
+    # return the instance of order
+    return order
+  end
+
+  def reset_cart
+    #reduce inventory for each product
+    to_remove = OrderItem.where(order_id: session[:order_id])
+    to_remove.each do |ordered_product|
+      remove_from = Product.find(ordered_product.product_id)
```

Comment:
This is a smidge confusing and I think it has to do with the name `remove_from`.
`remove` being a verb makes it hard for me to hold in my head the objects that we are working with.

Naming things is one of the hard problems in tech though. So sometimes that's just the way it goes.

One trick I use if I don't know what to call something is to at least make it noun-y if it is an object.
So maybe `removed_product` or `product_for_removal`